ecs/cluster/main.tf

```diff
@@ -1,61 +1,3 @@
-# ------------------------------------------------------------------------------
-# Variables
-# ------------------------------------------------------------------------------
-variable "prefix" {
-  description = "A prefix used for naming resources."
-}
-
-variable "vpc_id" {
-  description = "ID of the VPC for the subnets."
-}
-
-variable "subnet_ids" {
-  description = "ID of subnets where bastion can be provisioned."
-  type        = "list"
-}
-
-variable "instance_ami" {
-  description = "ID of a Amazon Linux ECS optimized AMI for the instances."
-  default     = "ami-1d46df64"
-}
-
-variable "instance_type" {
-  description = "Type of instance to provision."
-  default     = "t2.micro"
-}
-
-variable "instance_count" {
-  description = "Desired (and minimum) number of instances."
-  default     = "2"
-}
-
-variable "instance_key" {
-  description = "Name of an EC2 key-pair for SSH access."
-  default     = ""
-}
-
-variable "ecs_log_level" {
-  description = "Log level for the ECS agent."
-  default     = "info"
-}
-
-variable "ingress" {
-  description = "Map (port = source_security_group_id) which will be allowed to ingress the cluster."
-  type        = "map"
-  default     = {}
-}
-
-variable "ingress_length" {
-  description = "HACK: This exists purely to calculate count in Terraform. Should equal the length of your ingress map."
-  default     = 0
-}
```

Comment:
Couldn't this be solved with
```
"$(length(keys(ingress))"
```

Reply:
Think I've tried that already, and no dice. Thing is, the entire map is considered "calculated" if ANY of the values are references to other resources .... :( Can try again though.

Reply:
Thought that would work but if it doesn't, lets stick with this

Reply:
Still getting:
`module.cluster.aws_security_group_rule.ingress: aws_security_group_rule.ingress: value of 'count' cannot be computed`

However, this has got me thinking - when using a map we cannot have duplicate keys (?) and if we have multiple ALB's we'll likely want multiple "0" for the dynamic port mapping. I.e., perhaps we have to make the alb_arn into the key and the port into the value?

Once again, the use case I was thinking of when making it like this was concourse - since the ATC exposes two ports, I wanted the ability to route static ports. Should we instead assume that anything running on ECS only exposes 1 port, and is therefor dynamically routable, in which case we only need to the dynamic port range in all cases?

Reply:
Don't think we can get around `ingress_length`, but `ingress` can be made into a list of load balancers which will be authorized ingress to the cluster on the dynamic port range.

Reply:
There will presumable be at least one service exposing two ports for 80 -> 443 right?

Reply:
The container itself will expose one port, e.g. `8080` (HTTP): 
- Listener listens for either 80 or 443.
- Traffic on port 80 is passed to a target group which targets 8080 on the container.
- 8080 is mapped to a random ephemeral port on the instance (in dynamic range). This is behavior is specified by putting `0` as the hostPort in the task definition.

In this example, the load balancer (via listener and target group) ingresses the instance on the dynamic port range. So the Load balancers SG needs to be able to ingress the cluster on this port range.

The case I'm wondering about, is when we don't specify `0` as the hostPort in the task definition, in which case the container is only reachable by targeting that specific port on the instance. E.g., if we set `hostPort` to `8080`, the LB will target `8080` on the instance and require the SG to be allowed to ingress on that port.

In this last case, we'll have a port conflict if we try to launch multiple containers on the same instance (so we are limited to 1 per instance). However, this is how I had to do it to run the Concourse ATC in a container because it exposes 2 ports (8080 and 2222), but ECS cannot do dynamic port mapping for multiple ports.

So, I'm wondering if the use case I'm describing is rare enough that we don't need to support it in the module. I.e., we ONLY open the dynamic port range from the LB to the cluster. (Currently, the task definition is created inside the service module, and has the hostPort hardcoded to 0 anyway).

Reply:
Btw, I tested with a list now, but we still need `ingress_length` in order for it to work :x

Reply:
Now I understand you point and it makes sense. I think we can assume only dynamic ports, at least for now. If we need to fix it lets fix it later. @glennbech would this work for the the things you are looking at?

Reply:
The way I understood glenn, this should work - we can fix after weekend if I'm wrong 👍